File: sample/src/main/java/com/example/sample/MainActivity.java
package com.example.sample;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.widget.Toast;

import com.example.adaptablecurlpage.flipping.DynamicFlipView;
import com.example.sample.utils.ViewUtils;

import java.util.LinkedList;

public class MainActivity extends AppCompatActivity implements DynamicFlipView.OnPageFlippedListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
//        ViewUtils.hideSystemUI(this);
        setContentView(R.layout.activity_main);
        final LinkedList<String> pages_data = new LinkedList<>();
        for (int i = 0; i < 100; i++) {
            pages_data.add("okay " + i);
        }
        DynamicAdapter dynamicAdapter = new DynamicAdapter(pages_data, new DynamicAdapter.FlipBookAdapterCallbacks() {
            @Override
            public void onPageClicked(int pos) {
                Toast.makeText(MainActivity.this, "page " + pos + " is clicked", Toast.LENGTH_SHORT).show();
            }
        });
        final DynamicFlipView dynamicFlipView = findViewById(R.id.dynamic_flip_view);
        dynamicFlipView.setAdapter(dynamicAdapter);
        dynamicFlipView.setOnPageFlippedListener(this);
    }

    @Override
    public void onPageFlipped(View page, int page_no, long id) {

    }

    @Override
    public void onPageFlippedBackward(View page, int page_no, long id) {

    }

    @Override
    public void onPageFlippedForward(View page, int page_no, long id) {

    }

    @Override
    public void onUserFinger_isDown(View page, int page_no) {
//        ViewUtils.showSystemUI(this);
    }

    @Override
    public void onUserFinger_isUp(View page, int page_no) {
//        Runnable hide_runnable = new Runnable() {
//            public void run() {
//                ViewUtils.hideSystemUI(MainActivity.this);
//            }
//        };
//        new Handler().postDelayed(hide_runnable, 1000);
    }

    @Override
    public void onFastFlippingStarted(View page, int page_no, boolean is_forward) {
    }

    @Override
    public int onChangeActiveIndex(int active_index) {
        return 0;
    }

    @Override
    public void onFastFlippingEnded(View page, int page_no, boolean is_forward) {

    }

    @Override
    public void onRefreshForwardBackwardFinished(int time_taken) {

    }


    @Override
    public void onImageLoadingStatus(View page, int page_no, boolean is_ViewFullyLoaded) {

    }

    @Override
    public void onAfterViewLoadedToFlip(int page_no) {

    }

    @Override
    public void onRestorePage(int page_no, boolean is_forward) {

    }

    @Override
    public void onResetFlipSpeed() {

    }

    @Override
    public void onHold(View v, int pos) {

    }

    @Override
    public void onRelease(View v, int pos) {

    }

    @Override
    public void onEndPageReached(View v, int pos, boolean flip_forward) {

    }
}

File: sample/src/main/java/com/example/sample/activities/MainActivity.kt
package com.example.sample.activities

import android.content.Intent
import android.hardware.SensorDirectChannel
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.example.adaptablecurlpage.flipping.enums.FlipSpeed
import com.example.sample.R
import com.example.sample.activities.java.custom.MultiLayoutActivity
import com.example.sample.activities.java.custom.SingleLayoutActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun onMultiLayoutClick(view: View?) {
//        val intent = Intent(this, com.example.sample.activities.java.custom.MultiLayoutActivity::class.java)
//        val intent = Intent(this, com.example.sample.activities.java.simple.MultiLayoutActivity::class.java)
        val intent = Intent(this, com.example.sample.activities.kotlin.MultiLayoutActivity::class.java)
        startActivity(intent)
    }

    fun onSingleLayoutClick(view: View?) {
//        val intent = Intent(this, com.example.sample.activities.java.custom.SingleLayoutActivity::class.java)
//        val intent = Intent(this, com.example.sample.activities.java.simple.SingleLayoutActivity::class.java)
        val intent = Intent(this, com.example.sample.activities.kotlin.SingleLayoutActivity::class.java)
        startActivity(intent)
    }

    fun onFastFlip(view: View) {
        val intent = Intent(this, com.example.sample.activities.kotlin.MultiLayoutActivity::class.java)
        intent.putExtra("speed", FlipSpeed.VERY_FAST)
        startActivity(intent)
    }

    fun onSlowFlip(view: View) {
        val intent = Intent(this, com.example.sample.activities.kotlin.SingleLayoutActivity::class.java)
        intent.putExtra("speed", FlipSpeed.VERY_SLOW)
        startActivity(intent)
    }
}